### ISSUE TITLE
S2589 False positive with try/catch

### ISSUE THREAD

**Issue by antrv** [SonarSource/sonar-dotnet#2765]
S2589 False Positive. The issue is in try/catch statement.

```csharp
object obj = null;
try
{
    obj = GetObject(); // the method that can throw
    DoSomething(obj.ToString()); // this is important line, obj is dereferenced
}
catch
{
}

if (obj != null) // false positive S2589 here
{
    // do something
}
```

* SonarC# analyzer 8.0.0.9566
* Visual Studio Version 2017.3.17
* If running through the Scanner for MSBuild, its version

**Comment by pavel-mikula-sonarsource** [comment]
I'm closing this one, as it's already reported in #2761
